ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -15,52 +17,31 @@ using namespace std;
 using namespace std::chrono;
 using namespace std_msgs;
 
-const string FAULT_NAME_HPD           = "high_power_draw";
-const string FAULT_NAME_HPD_ACTIVATE  = "activate_high_power_draw";
-const int CUSTOM_FILE_EXPECTED_COLS           = 4;
-
-// Error flags.
-const int ERR_CUSTOM_FILE_FORMAT              = -1;
-
-// The index use to access temperature information.
-// This might change to median SOC or RUL index or fixed percentile.
-//
-static constexpr int TEMPERATURE_INDEX = 1;
-
-PowerSystemNode::PowerSystemNode()
-{ }
-
+/*
+ * Called on creating a PowerSystemNode object. Sets up class variables
+ * and initializes various other things.
+ */
 bool PowerSystemNode::Initialize()
 {
   if (!loadSystemConfig()) {
-    ROS_ERROR("Failed to load ow_power_system system config.");
+    ROS_ERROR("Failed to load PowerSystemNode system config.");
     return false;
   }
 
-  m_power_values.resize(m_moving_average_window);
-  std::fill(m_power_values.begin(), m_power_values.end(), 0.0);
-
-  if (!initPrognoser()) {
-    ROS_ERROR("Failed to initialize power system prognoser.");
-    return false;
-  }
-
-  if (!initTopics()) {
-    ROS_ERROR("Failed to initialize power system topics.");
-    return false;
-  }
+  m_init_time = system_clock::now(); // Taken from initPrognoser()
```

Comment:
``generateVoltageEstimate`` is called every cycle to update the voltage value sent into GSAP's prognosers. To my understanding it's supposed to mimic a sensor while using a random distribution. The minimum and maximum values are based off of the base voltage (``m_base_voltage``), the expected battery lifetime (``m_battery_lifetime``), and how long the simulation has been running (``timelapse``). But as you said, ``timelapse`` is based on ``system_clock::now()`` instead of ``ros::Time::now()``. I wouldn't know why, except for the simple explanation that it might just be a mistake and should be corrected.

I could easily change it, but should probably confirm with Chetan first to be safe.